project3.cc

```diff
@@ -0,0 +1,599 @@
+ */
+void ThreadedBinarySearchTree::insert(string key, int value)
+{
+  //Deal with the case that are first item is the header
+  if (empty())
+  {
+    Node *newnode = new Node();
+    _header->_lchild = newnode;
+
+    newnode->_lchild = makeThread(_header);
+    newnode->_rchild = makeThread(_header);
+    newnode->_value = value;
+    newnode->_key = key;
+  }
+  else
+  {
+    ThreadedBinarySearchTree::insertr(key, value, _header->_lchild);
+  }
+}
+
+
+/*
+ * Recursive insertion called by insert()
+ * PARAM-key: the string used to determine tree position
+ * PARAM-value: value that for the node
+ * PARAM-root: the starting node for each recursion
+ */
+void ThreadedBinarySearchTree::insertr(string key, int value, Node *& root)
+{
+  if (isThread(root->_lchild) && (key < root->_key))
+  {
+    Node *newnode = new Node();
+
+    newnode->_lchild = root->_lchild;
+    newnode->_rchild = makeThread(root);
+
+    root->_lchild = newnode;
+
+    newnode->_value = value;
+    newnode->_key = key;
+  }
+  else if (isThread(root->_rchild) && (key > root->_key))
+  {
+    Node *newnode = new Node();
+
+    newnode->_lchild = makeThread(root);
+    newnode->_rchild = root->_rchild;
+
+    root->_rchild = newnode;
+
+    newnode->_value = value;
+    newnode->_key = key;
+  }
+  else
+  {
+    if (key > root->_key)
+    {
+      insertr(key, value, root->_rchild);
+    }
+
+    else if (key < root->_key)
+    {
+      insertr(key, value, root->_lchild);
+    }
+  }
+}
+
+
+void ThreadedBinarySearchTree::erase(Iterator iter)
+{
+  // if (empty())
+  // {
+  //         return end();
+  // }
+  // else
+  // {
+  //         Node *tempnode;
+  //         Node *parent = this->parent()._ptr;
+  //         Iterator it;
+  //         for (iter = this->inorder_begin(); iter != this->end(); it++)
+  //         {
+  //                 if (key == iter._ptr->_key)
+  //                 {
+  //                         tempnode = iter._ptr;
+  //
+  //                         if(!isThread(tempnode->_rchild) && isThread(tempnode->_lchild))
+  //                                 delete iter._ptr;
+  //                         else if (isThread(tempnode->_rchild) && !isThread(tempnode->_lchild))
+  //                                 delete iter._ptr;
+  //                         else if(isThread(tempnode->_rchild) && isThread(tempnode->_lchild))
+  //                                 delete iter._ptr;
+  //                         else
+  //                                 delete iter._ptr;
+  //
+  //                 }
+  //         }
+  //         return end();
+  // }
+  return end();
+}
+
+
+/*
+* Determine if the tree is empty
+* RETURN-true: if there is only a header
+* RETRUN-false: if there is atleast one node
+*/
+bool ThreadedBinarySearchTree::empty() const
+{
+  if (makePointer(_header->_lchild) == _header)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+/*
+* Lookup function called the user clicks l
```

Comment:
Huh? (What does "called the user clicks l" mean?)
I think you mean it's called when they type L, but comments here should be about what it does, not about who calls it.  (There's no way to keep "who calls it" up to date, and it shouldn't matter.)